### DISCUSSION TITLE
enable_block_picking using Trame

### DISCUSSION THREAD

**Discussion by oscarBarEs** [Kitware/trame#494] [General] [answered]
Hi, for my project I want to have a similar code as the example of composite picking from pyvista:
https://docs.pyvista.org/version/stable/examples/02-plot/composite-picking.html
However, it doesnt seem to work. Maybe I need to modify vtkTrame.VtkLocalView but I been searching and cant find what exactly.

Heres my code: 

```
def make_poly():
    """Create a superellipsoid in a random location."""
    poly = pv.ParametricSuperEllipsoid(
        n1=np.random.random(),
        n2=np.random.random() * 2,
        u_res=50,
        v_res=50,
    )
    poly.points += np.random.random(3) * 20
    return poly
PYVISTA_GALLERY_FORCE_STATIC = True


# Assemble the multiblock and plot it using the default plotting settings
blocks = pv.MultiBlock([make_poly() for _ in range(100)])

pl = pv.Plotter()
actor, mapper = pl.add_composite(blocks, color="w", pbr=True, metallic=True)

for i in range(int(blocks.length)):
    mapper.block_attr[i].color = "r"


def callback(index, *args):
    """Change a block to red if color is unset, and back to the actor color if set."""
    if mapper.block_attr[index].color is None:
        mapper.block_attr[index].color = "r"
    else:
        mapper.block_attr[index].color = None


pl.enable_block_picking(callback, side="left")
pl.background_color = "w"
# pl.show()

server = get_server()
state, ctrl = server.state, server.controller

with SinglePageLayout(server) as layout:
    with layout.content:
        with vuetify.VContainer(
            fluid=True, classes="pa-0 fill-height", style="position: relative;"
        ):

            with vtkTrame.VtkLocalView(pl.ren_win)as local:
                def view_update(**kwargs):
                    local.update(**kwargs)                
                ctrl.view_update = view_update  

server.start()   
```

**Comment by jourdain** [comment, marked as answer]
Replace `VtkLocalView` by `VtkRemoteView`. 

Since the VtkLocalView is doing the rendering on the client side, if you don't do any extra work to sync the interaction (picking) with the server, none of what you are looking for is going to happen. You can find some picking example [here](https://github.com/Kitware/trame-vtk/blob/master/examples/validation/PickingLocalView.py).

**Reply by oscarBarEs** [reply]
Hi!
Yeah, by replacing it with remote it now works!

But with local view I have a problem and is that the function "vtk-local-view.get_scene_object_id is missing", so i could not run the example.

I went to the trame doc and i couldnt find that func (https://trame.readthedocs.io/en/latest/index.html) 
Maybe the name has changed?

**Reply by jourdain** [reply]
That method is right there https://github.com/Kitware/trame-vtk/blob/master/trame_vtk/widgets/vtk/common.py#L650

**Reply by jourdain** [reply]
That was added in `trame-vtk>=2.8.0`

**Reply by oscarBarEs** [reply]
yeah, you are right i was on 2.6.2. 

So, after updating I got this error when selecting an option on the Select:

 JS Error => TypeError: Cannot set properties of undefined (setting 'trame__local_view_1').
 
 So i checked and update trame and vtk as well. And now I have this versions 
```
Name: trame-vtk
Version: 2.8.5

```
```
Name: trame
Version: 3.6.0

```
```
Name: vtk
Version: 9.3.0

```
And now when running the code PickingLocalView.py the page doesnt open.
![image](https://github.com/Kitware/trame/assets/106106741/404f737d-671a-4299-a132-cccaedcb0cef)

Uncaught (in promise) TypeError: Cs is not a function
    at Os (index-26129a2a.js:23:3002)
    at index-26129a2a.js:23:4339


Btw, thank you for remanding me to update the libraries because now my main project works a lot faster.

**Reply by jourdain** [reply]
You may want to clear your browser cache.

**Reply by oscarBarEs** [reply]
As always, you are right.

Thanks a lot for taking the time!
